src/legacy/core_plugins/visualizations/public/np_ready/public/plugin.ts

```diff
@@ -110,12 +117,28 @@ export class VisualizationsPlugin
     setIndexPatterns(data.indexPatterns);
     setFilterManager(data.query.filterManager);
 
+    this.savedVisualizationDependencies = {
+      savedObjectsClient: core.savedObjects.client,
+      indexPatterns: data.indexPatterns,
+      chrome: core.chrome,
+      overlays: core.overlays,
+      visualizationTypes: types,
+    };
+
     return {
       types,
+      getSavedVisualizationsLoader: () => this.getSavedVisualizationsLoader(),
     };
   }
 
   public stop() {
     this.types.stop();
   }
+
+  private getSavedVisualizationsLoader = () => {
+    if (!this.savedVisualizations) {
+      this.savedVisualizations = createSavedVisLoader(this.savedVisualizationDependencies!);
```

Comment:
why not create this in start method and make it directly available on the contract ? (instead of providing this getter)

Reply:
It's a question of taste, but I think it's better to create this on demand, once it's needed. When you bootstrap Kibana, it runs all the `setup` and `start` functions. So I think it's good practice to run code that's not needed by all plugins on demand. `createSavedVisLoader` ain't a heavy function, so this will bring no performance improvements. However, it's just a pattern I've followed to be sure that no future changes of e.g. createSavedVisLoader add time to our global booting up process.